apps/api/v2/src/ee/bookings/2024-08-13/services/bookings.service.ts

```diff
@@ -990,4 +999,79 @@ export class BookingsService_2024_08_13 {
       t: await getTranslation("en", "common"),
     });
   }
+
+  async updateBooking(bookingUid: string, body: UpdateBookingInput_2024_08_13) {
+    const existingBooking = await this.bookingsRepository.getByUid(bookingUid);
+    if (!existingBooking) {
+      throw new NotFoundException(`Booking with uid=${bookingUid} not found`);
+    }
+    const { location } = body;
+
+    if (!location) {
+      return this.getBooking(existingBooking.uid);
+    }
+
+    return await this.updateBookingLocation(existingBooking, location);
+  }
+
+  async updateBookingLocation(existingBooking: Booking, location: BookingInputLocation_2024_08_13) {
+    const bookingUid = existingBooking.uid;
+    let bookingLocation = "";
+
+    if (existingBooking.userId === null) {
```

Comment:
I would write
```
if (!existingBooking.userId)
```

instead of explicit null checks here and also in other places just in case undefined has been stored as a value but thats just my opinion - technically should be fine.